### ISSUE TITLE
Animated Stickers

### ISSUE THREAD

**Issue by vitorhenrique123** [CrackerKSR/react-native-whatsapp-animated-stickers#1]
## Environment
System:
    OS: Linux 5.10 Manjaro Linux
    CPU: (12) x64 AMD Ryzen 5 5600X 6-Core Processor
    Memory: 651.52 MB / 15.61 GB
    Shell: 5.9 - /usr/bin/zsh
  Binaries:
    Node: 18.3.0 - /usr/bin/node
    Yarn: 1.22.17 - /usr/bin/yarn
    npm: 8.1.4 - /usr/local/bin/npm
    Watchman: Not Found
  SDKs:
    Android SDK: 31
  IDEs:
    Android Studio: Not Found
  Languages:
    Java: 1.8.0_332 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.0.0 => 18.0.0 
    react-native: 0.69.1 => 0.69.1 
  npmGlobalPackages:
    *react-native*: Not Found

## Describtion
@CrackerKSR Is support for animated stickers currently working correctly?
I'm trying to do some tests and unfortunately, it doesn't work when the animated_sticker_pack option is true. Also - debugging the DownloadImageAnimated module, but it seems incomplete. I didn't understand correctly how the animated stickers work.
![image](https://user-images.githubusercontent.com/18515574/178132838-db46d133-4268-48c3-94f7-b8f89e8562ac.png)
This only happens with animated stickers. Statics work very well - great job, by the way. 

## Reproducible Demo
JS
`    let pack = {
      "identifier": "animatedstickerstest",
      "name": "StickersAnimated",
      "publisher": "VitorTeste",
      "tray_image_file": "https://vitorhenrique.me/STK-20220705-WA0020.webp",
      "publisher_email": "contact@myproject.com",
      "publisher_website": "https://myproject.com",
      "privacy_policy_website": "https://myproject.com/legal",
      "license_agreement_website": "https://myproject.com/license",
      "animated_sticker_pack" : true,
      "avoid_cache" : true,
      "image_data_version": "1",
      "stickers": [
        {
          "image_file": "https://vitorhenrique.me/STK-20220705-WA0020.webp",  
          "emojis": ["✌️"]
        },
        {
          "image_file": "https://vitorhenrique.me/STK-20220705-WA0020.webp",
          "emojis": ["😍","😻"]
        },
        {
          "image_file": "https://vitorhenrique.me/STK-20220705-WA0020.webp",
          "emojis": ["😎"]
        }
      ]
    }`

**Comment by CrackerKSR** [comment]
Yes it is working properly. But I did not organised the code yet.

Total duration for sticker must not exceed 10 Sec
Each frame must be at least 8ms
The file size of the animated sticker must be less than 500KB
The sticker file must be Animated WEBP file. 
Image size must be 512x512 
Pack icon size 96x96
and I forgot the icon image file size limit.


I checked one of your sticker (https://vitorhenrique.me/STK-20220705-WA0020.webp)
It has 3.3ms Delay for each frame while it should be greater than or equal to 8ms

If you are making sticker app then you should verify each sticker file.
I will add sticker validation codes later.

**Comment by KishorJena** [comment]
It should work with local images.
it won't work for animated sticker as I had removed those codes which downloads image from url.

**Comment by vitorhenrique123** [comment]
Do you have any clue why it only works for local files?

> It should work with local images. it won't work for animated sticker as I had removed those codes which downloads image from url.

**Comment by KishorJena** [comment]
> Do you have any clue why it only works for local files?
> 
> > It should work with local images. it won't work for animated sticker as I had removed those codes which downloads image from url.

first of all your image is not meeting with requirements. I have tried with url t was working with proper image files. But later I had made many changes and I am not sure if Downloading image code is changed or not due to time gap.

I will fix it soon. There is another issue there adding 2 new sticker pack make changes in current pack. I fixed that too but did not push. I may create new repo after finalising.
I have added useful methods like resizing both animated and static images as per the requirements by whatsapp. Currently I am trying to make stickerpack editable. Means even after adding sticker to whatsapp user can add more stickers to exisintg pack or remove any sticker.

**Comment by KishorJena** [comment]
> Do you have any clue why it only works for local files?
> 
> > It should work with local images. it won't work for animated sticker as I had removed those codes which downloads image from url.

Okay I was working with same file today and found that I had removed URL(sUrl).openStream() for Animated. So it should download the static image files for now but now animated files.

**Comment by KishorJena** [comment]
try [react-native-whatsapp-stickers-manager](https://www.npmjs.com/package/react-native-whatsapp-stickers-manager)

**Comment by hyotam** [comment]
> try [react-native-whatsapp-stickers-manager](https://www.npmjs.com/package/react-native-whatsapp-stickers-manager)

@KishorJena what happened to that repo?

also is there a way to load images from urls on android and ios?

**Comment by KishorJena** [comment]
> > try [react-native-whatsapp-stickers-manager](https://www.npmjs.com/package/react-native-whatsapp-stickers-manager)
> 
> @KishorJena what happened to that repo?
> 
> also is there a way to load images from urls on android and ios?

Will be live soon. After done with fixing everything I will add url support for Android. ios is not supported yet for any operations.
